toner.go

```diff
@@ -0,0 +1,45 @@
+// tones generator
+
+
+package beep
+
+import "math"
+
+// simple sinusoid tone generator
+type toneStreamer struct {
+stat float64
+delta float64
+}
+
+// create streamer which will produce infinite sinusoid tone with the given frequency
+// use other wrappers of this package to change amplitude or add time limit
+// sampleRate must be at least two times grater then frequency, otherwise this function will panic
+func SinTone(sr SampleRate, freq int) Streamer {
+if int(sr) / freq < 2 {
+panic("samplerate must be at least 2 times grater then frequency")
```

Comment:
Instead of panic-ing when the SampleRate is not at least double the frequency, can you instead return an error so that the caller can handle the failure more easily.